### ISSUE TITLE
Can Cloak utilize more than one CPU?

### ISSUE THREAD

**Issue by burbilog** [cbeuw/Cloak#195]
I'm building a system facing several ISPs for multiple endusers and I have a question -- can Cloak handle multiple CPUs in single process? Does it make any sense to create multiple separate processes for each ISP (or even several listening ports on each ISP's IP, since my system has a lot of CPUs) or Cloak can handle different incoming connections on different CPUs within single process?

**Comment by cbeuw** [comment]
Yes. Like most idiomatic Go programs, Cloak uses goroutines liberally. All connections are handled with a dedicated goroutine so it'll utilise all your CPUs as much as possible

**Comment by hawshemi** [comment]
What are the softwares that Cloak is supported?
